src/util/ThreadAnnotations.h

```diff
@@ -0,0 +1,252 @@
+#pragma once
+
+// Copyright 2025 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+// Thread safety annotation macros for use with Clang's thread safety analysis
+// feature. These annotations allow the compiler to warn about potential thread
+// safety issues at compile time.
+//
+// Documentation for the annotations is available at:
+// https://clang.llvm.org/docs/ThreadSafetyAnalysis.html
+
+#ifndef THREAD_ANNOTATIONS_H_
```

Comment:
Given that this looks like a copy past from clang, this might belong in `lib` more so than `util` and may require a license entry

Reply:
This copies some of the _example_ provided in the documentation, https://clang.llvm.org/docs/ThreadSafetyAnalysis.html#mutex-h, with custom implementations of Mutex, MutexLocker etc. Seems weird to have a license for modified example code. Browsing other established repos, e.g. https://github.com/facebook/hermes/blob/0d66a2c3342e0722ef999087c6684cbfcfa099ab/API/hermes/ThreadSafetyAnalysis.h#L11,  there doesn't seem to be any mentions of licensing, just a link to the docs (which is already the case in this PR)